### ISSUE TITLE
Add support back for LMs which generate a different feature than the input ones

### ISSUE THREAD

**Issue by dpressel** [dpressel/mead-baseline#602]
We used to support (until very late in the 2.x branch), Kim et al 2015 (also ELMo) models as LMs, where we used one set of features for the src embeddings and one for the destination embeddings.  In the case of the Kim model specifically, we are tasked with generating the words, but the input features are characters, which makes the model very efficient for fine-tuning since it has many fewer parameters and the final softmax LM head is removed.

This was broken during the refactoring.  Its quite straightforward to add back the support -- I think we just need a custom `EmbeddingsStack` that can mask the input.

**Comment by dpressel** [comment]
This is probably not a good approach, since downstream LM fine-tuning would suffer from a near unusable interface.  Probably the better approach is to pass the `embeddings` dict into the `init_output()` method as the first argument, and simply remove it from the `EmbeddingsStack`.  Downstream applications with no LM head would then have a sensible set of keys to reload
